### PR TITLE
Issue/authentication

### PR DESCRIPTION
**Pull request by bashleigh**

### CODE REVIEW

File: src/auth.service.ts [hunk 1]
```diff
@@ -38,11 +38,11 @@ export default class AuthService {
 	}
 
 	async auth(auth: Auth): Promise<object> {
-		const user = await this.userService.findByEmail(auth.email);
+		const user = await this.userService.findByEmailWithPassword(auth.email);
 
 		if (!user) throw new BadRequestException('Invalid credentials');
-
-		const valid = this.userService.compareHash(auth.password, user.password);
+
```
Comment (S33G) [inline src/auth.service.ts]: Spaces!

File: src/auth.service.ts [hunk 2]
```diff
@@ -38,11 +38,11 @@ export default class AuthService {
 	}
 
 	async auth(auth: Auth): Promise<object> {
-		const user = await this.userService.findByEmail(auth.email);
+		const user = await this.userService.findByEmailWithPassword(auth.email);
 
 		if (!user) throw new BadRequestException('Invalid credentials');
```
Comment (S33G) [inline src/auth.service.ts:43]: This might be better if you used `UnauthorizedException`?
https://docs.nestjs.com/exception-filters

Comment (S33G) [reply src/auth.service.ts:43]: Or ForbiddenException

**Review by S33G — APPROVED** [review]

File: src/auth.service.ts (hunk 2 above)
Comment (bashleigh) [reply src/auth.service.ts:43]: User error = 400 bad response. Want to keep unauthorised and forbidden for their exclusive use only.

**Pull request merged**